src/cli/main.zig

```diff
@@ -781,18 +938,26 @@ pub fn setupSharedMemoryWithModuleEnv(gpa: std.mem.Allocator, roc_file_path: []c
 
     const shm_allocator = shm.allocator();
 
-    // Allocate space for the offset value at the beginning
-    const offset_ptr = try shm_allocator.alloc(u64, 1);
-    // Also store the canonicalized expression index for the child to evaluate
-    const expr_idx_ptr = try shm_allocator.alloc(u32, 1);
+    // Create a properly aligned header structure
+    const Header = struct {
+        parent_base_addr: u64,
+        entry_count: u32,
+        _padding: u32, // Ensure 8-byte alignment
+        def_indices_offset: u64,
+        module_env_offset: u64,
+    };
```

Comment:
I don't know if this was necessary in the end... but I spent a long time iterating on designs to fix alignment issues (probably of my own doing) and this seems like an improvement to leave here.